packages/flutter_tools/lib/src/base/file_system.dart

```diff
@@ -115,6 +108,13 @@ class FileSystemUtils {
   }
 }
 
+/// Return a relative path if [fullPath] is contained by the cwd, else return an
+/// absolute path.
+String getDisplayPath(String fullPath, FileSystem fileSystem) {
```

Comment:
This is fairly simple and only used in the resident runners and tests, I lifted it out to avoid needing to inject fs utils.